### ISSUE TITLE
Need More Helpful Error Message - HTTP ERROR 404

### ISSUE THREAD

**Issue by Pacius** [NebulousLabs/Sia-GPU-Miner#44]

**Comment by DavidVorick** [comment]
Which version of `siad`? Are you running `siad`?

**Comment by Pacius** [comment]
Now 1.1,no problem.But can this miner use  "siananopool"?And if,how can i set by ".bat"

**Comment by philipmw** [comment]
I am having this problem as well:

    $ ./sia-gpu-miner
    Initializing...

    HTTP error 404
    HTTP error 404
    HTTP error 404
    ...

I am using `siad` 1.1.1 with sia-gpu-miner 1.1.

From Wireshark, I see the failing HTTP request is `GET /miner/header HTTP/1.1`.

`siad` does not suggest that anything is wrong:

    $ ./siad
    Loading...
    (0/6) Loading siad...
    (1/6) Loading gateway...
    (2/6) Loading consensus...
    (3/6) Loading transaction pool...
    (4/6) Loading wallet...
    (5/6) Loading host...
    (6/6) Loading renter...
    Finished loading in 0.901766827 seconds
    Rescanning consensus set...
    Scanned to height 23316...
    Done!

**Comment by philipmw** [comment]
Ok, I figured it out. Apparently `siad` does not load the Miner module by default. Here's how I fixed this problem:

    ./siad -M cghmrtw

Then unlock the wallet:

    $ curl -i -A "Sia-Agent" -X POST localhost:9980/wallet/unlock?encryptionpassword=...

Now `sia-gpu-miner` works:

    $ ./sia-gpu-miner
    Initializing...

    Mining at 86.482 MH/ss	0 blocks mined

**Comment by DavidVorick** [comment]
That is correct, the miner module no longer loads by default as of v1.1.1 I believe.

Might be worth adding a more helpful error message though.

**Comment by DavidVorick** [comment]
I will actually leave this open in case someone wants to update the error message, but right now nobody is assigned to it.

**Comment by facialdeluxe** [comment]
Hi, I'm completly new to the sia mining, I apololgy if the question is too basic :
With Sia UI I managed to create a wallet and the blockchain downloaded 100%

Now, when I launch sia-gpu-miner.exe I (still) have error 404.

if in command prompt I type siad.exe it says :
Loading...
(0/6) Loading siad...
listen tcp 127.0.0.1:9980: bind: Une seule utilisation de chaque adresse de sock
et (protocole/adresse réseau/port) est habituellement autorisée.

siad -M cghmrtw
gives the same answer.

What should I do ?
Thanks !

**Comment by Zesseth** [comment]
Hi,

Here are instructions how I get it work:

1. Quit Sia-UI (using the Quit Sia option shown by right-clicking the notification tray icon) and try starting the siad agent directly.
For Windows: C:\Program Files (x86)\Sia-UI\resources\app\Sia\siad.exe --modules cgmhrtw -d %appdata%\Sia-UI\sia
NOTE: It may take a few minutes for Siad to finish loading, it took six for me with an SSD.
(Fix the path where SIA is placed / "installed")

2. Relaunch the Sia-UI (it will bind to your manually started siad instance)
3. Unlock your wallet
4. Run `miner start` command in the Terminal Plugin.

If it starts you can stop it running `miner stop` command.
I assume you get better results by running GPU miner and not "native CPU miner".
Of course you can run both at the same time if you like.

Now SIA-GPU-miner should work and you just to run it.
Leave command prompt open what you probably opened in step 1

Thanks for droghio for instructions in first place.
Original instructions can be found from here:
https://github.com/NebulousLabs/Sia-GPU-Miner/issues/59

**Comment by facialdeluxe** [comment]
Mmm closing and relaunching everything
siad -M cghmrtw
is loading the host 6/7

Don't know how long this should take...

jlahtela thank you !
Will try your method and I'll let you know !

Thank you !

**Comment by facialdeluxe** [comment]
A critical error loading Sia has occured: Siad unexpectedly exited. Error log: Loading...
(0/6) Loading siad...
listen tcp 127.0.0.1:9980: bind: Une seule utilisation de chaque adresse de socket (protocole/adresse réseau/port) est habituellement autorisée.

not good yet :/
running on a laptop with a gtx950M

**Comment by facialdeluxe** [comment]
I can't seem to launch both siad and sia-UI at the same time.

If I siad -M cghmrtw
sia-UI crashes

and therefore, if I don't start siad, sia UI starts well, but if I type "miner start" in the terminal,  it says : Could not start miner: API call not recognized: /miner/start

**Comment by mysteryr** [comment]
Hi guys, sorry but i'm new in sia mining. i cannot open sia gpu miner, do i need to wait about the end of the synchronization of Sia ? Can you tell me what he sync exactly? how long times will i have to wait about the end of this?   Thanks for your help

**Comment by droghio** [comment]
@facialdeluxe Sia-UI will try to launch a new instance of siad, if you then try to manually launch siad in a terminal it won't work since the other run is running an claimed the ports used to communicate with the outside world. If you fully quit the Sia-UI you should be able to follow the steps @jlahtela relinked, but it may take upwards of five minutes for the miner module to load.

@mysteryr You need to use a terminal to launch the miner atm, you can't just on the .exe. You should wait to synchronize until you start mining otherwise you will only be mining stale blocks (which aren't worth anything). To sync just leave Sia-UI up and running for a while. You should see a status bar in the top right giving you the progress of synchronization or if the block chain is synchronized. How long it takes will depend on your internet speed, but usually several hours. The block chain is about 4gb atm.

Hope that helps.

**Comment by facialdeluxe** [comment]
Woohoo I'm mining !

What does the " -d %appdata%\Sia-UI\sia" do ?

**Comment by facialdeluxe** [comment]
Question 👍 
The cpu is also mining, but the cpu activity is at 20%. 
Though it says "CPU Miner is now running."
it's either not working, or there is only one core mining.
Is there a command to make it mine at 100% ?
I'm aware cpu is not efficient compared to gpu, but just to learn...

Thank you :)

**Comment by facialdeluxe** [comment]
Other question,
the GTX 960M is mining between 13 and 22 MH/s
in 2 hours it seems I have zero block or sia coin, is this normal ?

**Comment by plussiminus** [comment]
Hello guys!

I just want to thank all of you, but specially @jlahtela for helping me run the SC miner.

Have a great day and good luck with everything!
Peace!

**Comment by SFoskett** [comment]
Here's how to get sia-gpu-miner working on Mac OS X:
1) Install and run Sia UI and wait for it to finish synchronizing. Create a wallet. Save the seed. Then quit the UI.
2) Download and run siad with the blockchain the UI just accessed from the directory that contains siad:
./siad -M cghmrtw -d "/Users/yourname/Library/Application Support/Sia-UI/sia"
3) Wait until siad is done loading ("Loading wallet..." will take a while) then run the UI again
4) Back in the UI, click Wallet and unlock it with your seed. This will take a few more minutes.
5) From the directory containing sia-gpu-miner, run this and it should work:
./sia-gpu-miner -C 3000 -I 16 -d 0
(adjust C to update about once per second, I to whatever you're comfortable with, and d for the proper GPU)

FWIW I'm getting around 200 MH/s on each of the AMD FirePro D300 GPU's in my 2013 Mac Pro at 16 intensity. I get almost 2300 MH/s at intensity 20!

**Comment by drewlustro** [comment]
Thank you @SFoskett.

My issue was that upon booting `siad` with the miner module, it would take forever to perform the `Loading miner` step (not loading wallet).... once I let that fully scan the blockchain, I had success. Also mining on dual 2013 Mac Pro FirePro D300's. Nice machine, mate.


---

<br>

![macpro](https://user-images.githubusercontent.com/194885/34342791-0d4d83a2-e970-11e7-8578-54942744efd8.jpg)
